sched/scheduler/stateful_scheduler.go

```diff
@@ -270,7 +265,8 @@ func (s *statefulScheduler) scheduleTasks() {
 	}
 
 	// Calculate a list of Tasks to Node Assignments & start running all those jobs
-	taskAssignments := getTaskAssignments(s.clusterState, unscheduledTasks)
+	taskAssignments, affinity := getTaskAssignments(s.clusterState, unscheduledTasks)
```

Comment:
We can revisit this later. The idea here is that task_scheduler needs to know all the tasks at once to do basic scheduling (or it needs to take additional params for # of tasks remaining with the same jobID, # of jobs, # of remaining tasks with different jobID, etc).